packages/fuels-core/src/code_gen/functions_gen.rs

```diff
@@ -65,6 +65,50 @@ pub fn expand_function(
     })
 }
 
+/// Generate the function (as a `TokenStream`) used to encode into script data the arguments to the
+/// `main` function of a deployed script.
+pub fn generate_script_main_function(
+    main_function_abi: &ABIFunction,
+    types: &HashMap<usize, TypeDeclaration>,
+) -> Result<TokenStream, Error> {
+    if main_function_abi.name != "main" {
+        return Err(Error::InvalidData(
+            "Script `main` function name can not be different from `main`".into(),
+        ));
+    }
+
+    let args = function_arguments(main_function_abi, types)?;
+
+    let arg_names = args.iter().map(|component| &component.field_name);
+
+    let arg_declarations = args.iter().map(|component| {
+        let name = &component.field_name;
+        let field_type: TokenStream = (&component.field_type).into();
+        quote! { #name: #field_type }
+    });
+
+    let doc = expand_doc(
+        "Encode the arguments provided so they can be passed as argument to the script's main \
+        function",
+    );
+
+    let name = safe_ident("main");
+
+    Ok(quote! {
+        #doc
+        pub async fn #name(&self #(,#arg_declarations)*) -> Result<Vec<Receipt>, SDKError> {
+            let provider = self.wallet.get_provider().expect("Provider not set up").clone();
+            let tokens = [#(#arg_names.into_token()),*];
+            let script_data = ABIEncoder::encode(&tokens)?.resolve(0);
```

Comment:
Resolving 0 here will not work for vectors, the proper loading address needs to be determined.

I'm for either implementing vector support as a script argument or failing in compile-time if somebody should attempt to call the script gen on such a script until such support is added.